judge/runguard.c

```diff
@@ -349,6 +351,7 @@ Run COMMAND with restrictions.\n\
   -r, --root=ROOT        run COMMAND with root directory set to ROOT\n\
   -u, --user=USER        run COMMAND as user with username or ID USER\n\
   -g, --group=GROUP      run COMMAND under group with name or ID GROUP\n\
+  -d, --chdir=DIR        change to directory DIR after setting root directory\n\
```

Comment:
why can't we switch directory before calling runguard?

Reply:
Because runguard does the chroot()ing. So if we chdir before we’re not in the dir with the right bind mounts.

Or put differently, the chroot is started once for a judging, which means both the compile and the run phase. So we use a different subdir for compile to not make it too messy